syntax = "proto2";

package leo;

message RequestVoteArgs {
    required uint32 term = 1;
    required int32 candidate = 2;
    required uint32 last_log_index = 3;
    required uint32 last_log_term = 4;
}

message RequestVoteReply {
    required uint32 term = 1;
    required bool vote_granted = 2;
}

message RequestAppendArgs {
	required uint32 term = 1;
	required int32 leader_id = 2;
	required uint32 pre_log_index = 3;
	required uint32 pre_log_term = 4;
	required uint32 leader_commit = 5;
}

message RequestAppendReply {
	required uint32 term = 1;
	required bool success = 2;
	required uint32 conflict_index = 3;
	required uint32 conflict_term = 4;
}